up42/asset.py

```diff
@@ -47,6 +47,15 @@ def info(self) -> Dict:
         self._info = response_json["data"]
         return response_json["data"]
 
+    @property
+    def source(self) -> Dict:
+        """
+        Gets the source of the Asset. One of `TASKING`, `ORDER`, `BLOCK`, `UPLOAD`.
+        """
+        source = self.info["source"]
+        logger.info(f"Asset source is {source}")
+        return source
```

Comment:
So these are the type of assets we would have in user storage right? But we only want the asses output by blocks to be reacheable via asset id according to line 354. Why is that? Is it because everything under TASKING, ORDER and UPLOAD hasn't been fully processed and finalised, hence is not available?

Reply:
No - the source of assets looks like it is static - that means it depends on how it was created. If an order comes from our order service it will have "ORDER" as source, if it comes from a previously run download block it will have "BLOCK" as source, regardless if it is completed or not. The issue here is that we do not support inputing all assets into workflows - assets that are not "BLOCK" source don't have a data.json and because of this we can not use them for now in a workflow. Hope this makes sense?

Reply:
Ahn! Makes a lot of sense! Thanks for clarifying :)

Reply:
Good to know, thanks for question & answer.